fvm/scriptEnv.go

```diff
@@ -660,70 +660,43 @@ func (e *ScriptEnv) ImplementationDebugLog(message string) error {
 	return nil
 }
 
-func (e *ScriptEnv) ProgramParsed(location common.Location, duration time.Duration) {
+func (e *ScriptEnv) RecordTrace(operation string, location common.Location, duration time.Duration, logs []opentracing.LogRecord) {
 	if e.isTraceable() {
-		locStr := ""
+		if logs == nil {
+			logs = make([]opentracing.LogRecord, 0)
+		}
 		if location != nil {
-			locStr = location.String()
+			logs = append(logs, opentracing.LogRecord{Timestamp: time.Now(),
+				Fields: []traceLog.Field{traceLog.String("location", location.String())},
+			})
 		}
-		e.ctx.Tracer.RecordSpanFromParent(e.traceSpan, trace.FVMCadenceParseProgram, duration,
-			[]opentracing.LogRecord{
-				{Timestamp: time.Now(),
-					Fields: []traceLog.Field{traceLog.String("location", locStr)},
-				},
-			},
-		)
+		spanName := trace.SpanName(fmt.Sprintf("%s.%s", trace.FVMCadenceTrace, operation))
```

Comment:
It might be good to avoid string formatting here and require callers to provide a constant string instead

Reply:
hmm, I think the idea here is to prefix the traces with the component path (e.g. everything under env is prefixed with 'fvm.env.xxx' and I was thinking everything from cadence runtime be prefixed with `fvm.cadence.xxx`. 

I was thinking to move this logic into the SpanName as a method called child which yields a new SpanName and hide this magic there. what do you think ? 

```
type SpanName string

func (s SpanName) Child(subOp string) SpanName {
	return SpanName(fmt.Sprintf("%s.%s", s, subOp))
}
```

Reply:
Yes, that makes sense, but I was worried about the cost of the formatting on each trace. Currently the names are constants

Reply:
I see, that's definitely a cost, though I think the cost might be worth it initially, I did change it to string concatenation which takes a few nano seconds instead. I thought even if we decide to do be passed as a constant string by the caller, caller probably has to build the string.